xwiki-platform-core/xwiki-platform-web/xwiki-platform-web-war/src/main/webapp/resources/uicomponents/extension/extension.js

```diff
@@ -495,6 +495,12 @@ XWiki.ExtensionBehaviour = Class.create({
       parameters : formData,
       onCreate : function() {
         dependency.removeClassName('extension-item-unknown').addClassName('extension-item-loading');
+        // Remove the unknown icon.
+        dependency.removeChild(dependency.childNodes[1]);
+        // Add the load icon
+        const loadPath = '$xwiki.getSkinFile("icons/xwiki/spinner.gif")';
+        const loadIcon = '<img  src="'+ loadPath + '" alt=""/>';
```

Comment:
This doesn't match the behavior of the spinner Velocity macro. A possible solution is to always output the spinner from ``extension.vm``, but hidden by default. It would be made visible by the ``extension-item-loading`` CSS class, which means no change in needed in this JavaScript file.